### PR TITLE
Ticket7589 install script fix adding autostart link

### PR DESCRIPTION
**Pull request by pheest**
If the autostart is in the 'all users' folder, it is removed from there.
If the spudulike user doesn't exist yet, it is created and file copied.
If the spudulike user exists, but the autostart isn't present there, it is copied in.
If the autostart is already where it should be, does nothing.

### CODE REVIEW

**Comment by FreddieAkeroyd** [comment]
We will probably need to separate the creation of spudulike out from the link creation a bit more, we don't want the account accidentally created on a developer computer testing an install. Also we have a few special cases like the engfin stress computer that do not run under spudulike and so would not want one creating. Normally an install would be run under the account that will be running ibex, so making links for "current logged on user" would cover things. The special case is that if an install is being done as part of an instrument image build, in which case it would want links in spudulike which may not exist at that point.

**Comment by pheest** [comment]
Added username verification on account creation.
If the user isn't 'Administrator', nothing is done.
So this can only happen in the special case of new computer install.

File: installation_and_upgrade/ibex_install_utils/tasks/system_tasks.py [hunk 1]
```diff
@@ -327,3 +360,4 @@ def user_confirm_upgrade_type_on_machine(self, machine_type):
         answer = self.prompt.prompt("Continue? [Y/N]", ["Y", "N"], "Y")
         if answer != "Y":
             raise UserStop()
+
```
Comment (NikolaRoev) [inline installation_and_upgrade/ibex_install_utils/tasks/system_tasks.py:363]:
```suggestion

```
Removing white space should fix the no new line indicator.

File: installation_and_upgrade/ibex_install_utils/tasks/system_tasks.py [hunk 2]
```diff
@@ -255,17 +256,49 @@ def put_autostart_script_in_startup_area(self):
...
+            
+        if os.path.exists(USER_HOME) and not os.path.exists(user_folder):
+            # The user was created but we (Administrators) don't have access to the folder.
+            admin_commands = AdminCommandBuilder()
+            admin_commands.add_command("icacls", USER_HOME + " /inheritance:e /grant:r Administrators:R")
+            admin_commands.run_all()
+
+        if not os.path.exists(USER_HOME):
+            myname = os.getlogin()
+            if (myname == "Administrator"):
```
Comment (NikolaRoev) [inline installation_and_upgrade/ibex_install_utils/tasks/system_tasks.py:278]:
```suggestion
            if myname == "Administrator":
```

**Review by NikolaRoev — CHANGES_REQUESTED** [review]
The ibex system boot batch file is currently added to start up only for the `spudulike` account. As mentioned in https://github.com/ISISComputingGroup/ibex_utils/pull/153#issuecomment-1441706421 some machines might not be using that user name.
I think this needs to be examined more to see if this is accepted behaviour or if the batch file should be added for the current user, or if the user should be decided by some other means. Was there an issue with adding the batch file to the global startup folder?

I think creation of the `spudulike` user should be separated out into its own task if it's still relevant. Someone doing a new install might not want to create a new user without knowing in the "autostart" step, even if the situation might be quite rare.

Something to keep in mind, if it was not known, `os.getlogin` will return the account that started the terminal not the currently logged in one.

Please revert the change done in https://github.com/ISISComputingGroup/ibex_utils/pull/145.

**Review by NikolaRoev — CHANGES_REQUESTED** [review]
If there is an error during the execution of `create_spudulike.bat` through the Python code and the new terminal is closed manually the initial terminal will hang.

Even if there is a failure in `create_spudulike.bat` that is not reflected in the install script and execution continues as if the task was successful.

I would recommend doing everything that is currently done in the batch file in the Python code and through the `AdminCommandBuilder` class.

File: installation_and_upgrade/ibex_install_utils/tasks/system_tasks.py [hunk 3]
```diff
@@ -253,9 +254,29 @@ def create_spudulike_user(self):
...
+            user_folder = os.path.join(USER_START_MENU, "Programs", "Startup")
+            # The 'runas' command requires the user to enter the password in response to a prompt.
+            # This isn't possible within the AdminCommandBuilder class becuase stdout and stderr are redirected to a file.
+            # So the user never sees the prompt.
             my_path = os.path.abspath(os.path.dirname(sys.argv[0]))
             admin_commands.add_command("start /wait", my_path + "\\create_spudulike.bat " + from_path)
+            # So doesn't ask is file name or directory
+            admin_commands.add_command("mkdir", " \"" + user_folder + " \"")
+            # Copy the startup file
+            admin_commands.add_command("xcopy", "/I " + from_path + " \"" + user_folder + "\"")
```
Comment (NikolaRoev) [inline installation_and_upgrade/ibex_install_utils/tasks/system_tasks.py:275]: Why are we copying the startup file in the `Create spudulike user` step?

**Review by Tom-Willemsen — APPROVED** [review]

**Pull request merged**